server/controller/dailySchedule.js

```diff
@@ -0,0 +1,82 @@
+import { check, validationResult } from "express-validator";
+// MODELS
+import DailySchedule from "../models/DailySchedule.js";
+import Chore from "../models/Chore.js";
+import User from "../models/User.js";
+
+/*
+ ** This function creates a new schedule for the day
+ ** @param {Object} req - The request object
+ ** @param {Object} res - The response object
+ */
+ export const createDailySchedule = async (req, res) => {
+
+    // Sanitize data
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+        // const timeframe = req.querry.timeframe; //extracting timeframe from request
+
+        var dailySchedule = []; // will add chores for the day in here
+
+        // get the user that is requesting the schedule
+        const userID = req.user.id;
+        const user = await User.findOne({ _id: userID }); // search for the user
+        const choreList = user.chores;
+        
+        console.log("Printing chore ids for this user")
+
+        // todays date in ms
+        const today = Date.now();
+
+        for (const choreId of choreList) {
+            // console.log(choreId);
+            const chore = await Chore.findOne({ _id: choreId });
```

Comment:
Surely there's a way to only query mongoDB once to get a dictionary of all of a user's chores, right? Looping through sends a request for every individual chore ID.